scripts/image-diff

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+
+# image-diff - Generate a diff of two images.
+# by Steven Elliott <selliott512@gmail.com>
+#
+# The images being diff'd are first converted to grey scale. The pixels with
+# brightness above a certain threshold are considered to be on, and the rest
+# off. In the resulting diff image pixel color is based on the following table:
+#
+#   image-1     image-2     diff-out
+#   -------     -------     --------
+#   on          on          white
+#   on          off         red
+#   off         on          blue
+#   off         off         black
+#
+# So blue is added and red is removed.
+#
+# The diff image has smaller dimension of the two images being diffed. If those
+# images have the same dimensions the so will the diff image.
+
+import os
+import sys
+
+# Constants
+
+# Colors that can be adjusted.
+color_both = (255, 255, 255) # white
+color_1    = (255,   0,   0) # red
+color_2    = (0  ,   0, 255) # blue
+color_none = (0  ,   0,   0) # black
+
+# Above this is considered to be white / on / true.
+threshold = 30
+
+# Functions
+
+def open_bw_image(name):
+    image = Image.open(name)
+    return image.convert("L")
+
+def is_on(pixel):
+    return pixel >= threshold
+
+from PIL import Image
+
+if len(sys.argv) != 4:
+    print >> sys.stderr, "Usage: image-diff image-1 image-2 diff-out"
```

Comment:
Indeed that does not work on Python 3. I'm trying to recall what my intent was for supporting different Python versions for this script. In any case, `wad2image`, which will replace this, was tested on Python 2 and 3.